### PR TITLE
Wireguard docs updates

### PR DESCRIPTION
**Pull request by robbrockbank**
## Description

Updates to the wireguard docs to remove "tech preview" classification, and to link to MTU doc.
Update to MTU doc to discuss MTU requirements for wireguard

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit 84b4ea3a6130172b4f2cc63ec7495886b1acc3ab

https://deploy-preview-3658--calico.netlify.app

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit 6ed3b81769d3733b884f4cfffee4c2e579f2aa43

https://deploy-preview-3658--calico.netlify.app

File: security/node-to-node-encryption.md [hunk 1]
```diff
@@ -28,30 +25,54 @@ This how-to guide uses the following {{site.prodname}} features:
...
+  ```
+  calicoctl patch felixconfiguration node.my-node --type='merge' -p '{"spec":{"wireguardEnabled":false}}'
+  ```
+
+Calico will not encrypt any of the pod traffic to or from this node.
+  
+### How to disable wireguard for the cluster
 
-To disable WireGuard on a specific node with WireGuard installed, modify the host-specific Felix configuration. For example:
+To disable WireGuard on all nodes modify the host-specific Felix configuration. For example:
```
Comment (realgaurav) [inline security/node-to-node-encryption.md:70]: remove `host-specific`

File: security/node-to-node-encryption.md [hunk 2]
```diff
@@ -28,30 +25,54 @@ This how-to guide uses the following {{site.prodname}} features:
...
+  calicoctl patch felixconfiguration node.<Node-Name> --type='merge' -p '{"spec":{"wireguardEnabled":false}}'
+  ```
+
+For example, to disable encryption for pod traffic on node "my-node", run the following command:
+
+  ```
+  calicoctl patch felixconfiguration node.my-node --type='merge' -p '{"spec":{"wireguardEnabled":false}}'
+  ```
+
+Calico will not encrypt any of the pod traffic to or from this node.
```
Comment (realgaurav) [inline security/node-to-node-encryption.md:66]: `any pod traffic`?

Comment (realgaurav) [reply security/node-to-node-encryption.md:66]: Note at :43 already mentions this. I think we can remove this line. WDYT?

File: networking/mtu.md [hunk 3]
```diff
@@ -47,51 +46,72 @@ The following table lists common MTU sizes for {{site.prodname}} environments. B
...
+| 1460 (GCE)             | 1460                  | 1440                                       | 1410                                    | 1400                                        |
+| 9001 (AWS Jumbo)       | 9001                  | 8981                                       | 8951                                    | 8941                                        |
+| 1450 (OpenStack VXLAN) | 1450                  | 1430                                       | 1400                                    | 1390                                        |
 
 **Recommended MTU for overlay networking**
 
-The extra overlay header used in IP in IP and VXLAN protocols, reduces the minimum MTU by the size of the header. (IP in IP uses a 20-byte header, and VXLAN uses a 50-byte header). Therefore, we recommend the following:
+The extra overlay header used in IP in IP, VXLAN and Wireguard protocols, reduces the minimum MTU by the size of the header. (IP in IP uses a 20-byte header, VXLAN uses a 50-byte header, and Wireguard uses a 57-byte header). Therefore, we recommend the following:
```
Comment (realgaurav) [inline networking/mtu.md:59]: s/57/60.

File: networking/mtu.md [hunk 4]
```diff
@@ -47,51 +46,72 @@ The following table lists common MTU sizes for {{site.prodname}} environments. B
...
 ```
 
-#### Configure MTU for overlay networking
+After updating the ConfigMap perform a rolling restart of all calico/node pods.
 
-If you are using IP in IP and/or VXLAN for {{site.prodname}} overlay networking, set the tunnel MTU to match the value that you configured for the veth MTU. 
+##### Openshift
 
-Edit `calico-config ConfigMap` to set the MTU tunnel values in FelixConfiguration. For example: 
+For Openshift, edit the {{site.prodname}} operator Installation resource to set the `mtu` field in the `calicoNetwork` section
```
Comment (caseydavenport) [inline networking/mtu.md:100]: Openshift -> "OpenShift"

File: networking/mtu.md [hunk 5]
```diff
@@ -47,51 +46,72 @@ The following table lists common MTU sizes for {{site.prodname}} environments. B
...
 
-```conf
-# Configure the MTU to use
-veth_mtu: "1440" 
+```bash
+oc patch installation.operator.tigera.io default --type merge -p '{"spec":{"calicoNetwork":{"mtu":1440}}}'
 ```
 
-#### eBPF mode
+##### Operator (non-Openshift)
```
Comment (caseydavenport) [inline networking/mtu.md:107]: Do we need separate openshift / non-openshift instructions for the operator? Can we just have one "Using the operator? (e.g., on openshift)" section?

File: security/node-to-node-encryption.md (hunk 1 above)
Comment (realgaurav) [reply security/node-to-node-encryption.md:70]: s/`host-specific`/`default`

File: networking/mtu.md (hunk 5 above)
Comment (realgaurav) [reply networking/mtu.md]: Let me rephrase this and see if it looks OK.

File: security/node-to-node-encryption.md (hunk 2 above)
Comment (realgaurav) [reply security/node-to-node-encryption.md]: changed this a bit to suit example command.

**Comment by bmckercher123** [comment]
I'm having similar issues trying to present the install methods in a way that users do not have to think. I'm not sure anyone would remember if the started a cluster using Quickstart. Does this work to differentiate?

For OS...use only Kubernetes and Openshift. 
- Kubernetes (this is manifest)
- OpenShift (this is operator)

For Enterprise when you merge it...
- Delete Kubernetes section entirely
- Keep Openshift section, but only if it is different from standard operator
- Get rid of any reference to operator

**Comment by realgaurav** [comment]
> 
> For OS...use only Kubernetes and Openshift.
> 
> * Kubernetes (this is manifest)
> * OpenShift (this is operator)
> 

Thanks @bmckercher123!

We can have _Operator_ based installation for non-_OpenShift_ deployments. With that logic, dividing between _Kubernetes_/_OpenShift_ can be confusing IMO. I am open to changing  currently proposed _Manifest_/_Operator_ sections.

File: _data/navbars/security.yml [hunk 6]
```diff
@@ -76,8 +76,8 @@ section:
     path: /security/high-connection-workloads
   - title: Defend against DoS attacks
     path: /security/defend-dos-attack
-- title: Try out the pod-to-pod encryption tech preview
-  path: /security/try-node-to-node-encryption
+- title: Pod-to-pod encryption
+  path: /security/pod-to-pod-encryption
```
Comment (lxpollitt) [inline _data/navbars/security.yml:80]: Also need to add an entry to `netlify/_redirects` file to redirect from the old URL to the new one.

File: _data/navbars/security.yml [hunk 7]
```diff
@@ -76,8 +76,8 @@ section:
     path: /security/high-connection-workloads
   - title: Defend against DoS attacks
     path: /security/defend-dos-attack
-- title: Try out the pod-to-pod encryption tech preview
-  path: /security/try-node-to-node-encryption
+- title: Pod-to-pod encryption
```
Comment (lxpollitt) [inline _data/navbars/security.yml:79]: Maybe "Encrypt in-cluster pod traffic" would be better?

File: networking/mtu.md [hunk 8]
```diff
@@ -37,65 +37,82 @@ For help using IP in IP and/or VXLAN overlays, see [Configure overlay networking
...
+
+For manifest based installations (i.e. ones that do not use the operator) edit the `calico-config` ConfigMap. For example:
 
 ```bash
 kubectl patch configmap/calico-config -n kube-system --type merge \
   -p '{"data":{"veth_mtu": "1440"}}'
 ```
 
-#### Configure MTU for overlay networking
+After updating the ConfigMap perform a rolling restart of all calico/node pods.
```
Comment (lxpollitt) [inline networking/mtu.md:96]: Several questions on this:
1) Can you help educate me on why this is needed? (I thought calico/node picks up config changes without requiring a manual restart usually?)
2) How does a user actually do this rolling restart?
3) Would they also need to restart all pods on each node so they get the right MTU?

File: security/pod-to-pod-encryption.md [hunk 9]
```diff
@@ -1,14 +1,11 @@
 ---
-title: Try out the pod-to-pod encryption tech preview
-description: Try out enabling WireGuard for state-of-the-art cryptographic security between pods for Calico clusters.
+title: Pod to pod encryption
```
Comment (lxpollitt) [inline security/pod-to-pod-encryption.md:2]: Suggest "Encrypt in-cluster pod traffic".  Might also change the path to be `security/encrypt-pod-traffic` or similar (which isn't quite as accurate but will likely score better with Google SEO).

File: security/pod-to-pod-encryption.md [hunk 10]
```diff
@@ -1,14 +1,11 @@
...
+description: Enable WireGuard for state-of-the-art cryptographic security between pods for Calico clusters.
 ---
 
 ### Big picture
 
-Enable WireGuard to secure on the wire pod-to-pod traffic in a {{site.prodname}} cluster.
-
-> **Warning!** Pod-to-pod encryption is a tech preview and should not be used in production clusters. It has had very limited testing and it will contain bugs (please report these on the Calico Users Slack or GitHub). This feature is currently not supported with overlay networks (IP in IP or VXLAN) due to known issues with NodePort services.
-{: .alert .alert-danger}
+Enable WireGuard to secure on the wire pod-to-pod pod traffic in a {{site.prodname}} cluster.
```
Comment (lxpollitt) [inline security/pod-to-pod-encryption.md:8]: Could re-phrase this too if we're going for encrypt in-cluster pod traffic.  i.e. replace "pod-to-pod" with "in-cluster pod"

File: networking/mtu.md (hunk 8 above)
Comment (realgaurav) [reply networking/mtu.md:96]: @lxpollitt -- the proposed statement is just the rephrasing the original text. I am not sure if anything has change since the original. I will check if the restart is a must.

```
When you set the MTU, it applies to new workloads. To apply MTU changes to existing workloads, you must restart calico nodes. Restarting the calico/node pods takes values from the ConfigMap and starts rolling updates for any {{site.prodname}} tunnel network interfaces on the node. 
```

Comment (realgaurav) [reply networking/mtu.md:96]: > Can you help educate me on why this is needed? (I thought calico/node picks up config changes without requiring a manual restart usually?)

There is no code watching update to calico-config. I see some code looking for changes to _token_, _root-ca file_ but not for other configuration. This is why restart is required.

> How does a user actually do this rolling restart?
> Would they also need to restart all pods on each node so they get the right MTU?

Running command like `kubectl rollout restart daemonset calico-node -n kube-system` should restart all calico-node pods. I will add an example command in the document to make it clearer.
